### PR TITLE
Restoring error object from message

### LINKED ISSUE #83: Error is lost during in the outgoing message
The problem lies [here](
https://github.com/dial-once/node-bunnymq/blob/develop/src/modules/message-parsers.js#L29)
The reason for that is because JSON.stringify(error) results in an empty object.

Suggested solution is to use [this module](https://www.npmjs.com/package/deserialize-error)

Test case
```js
// inside a consumer handler
.catch((error) => {
  return { error };
});
```

```js
producer.produce(<queueue>, <message>, { rpc: true })
.then((response) => {
   console.log(response);   // { error: {} }
});
```

### PR DESCRIPTION
**Pull request by Spring3**
Resolves #83

### CODE REVIEW

**Review by mrister — APPROVED** [review]
LGTM we just need to see why are the tests failing on CIRCLE_CI

File: src/modules/message-parsers.js [hunk 1]
```diff
@@ -6,7 +8,11 @@
 module.exports.in = (msg) => {
   // if sender put a json header, we parse it to avoid the pain for the consumer
   if (msg.properties.contentType === 'application/json') {
-    return JSON.parse(msg.content.toString());
+    const content = JSON.parse(msg.content.toString());
```
Comment (mrister) [inline src/modules/message-parsers.js]: maybe oyu can adda check to see if content is set on message so toString() does not fail. Basically like here https://github.com/dial-once/node-bunnymq/pull/85/files#diff-e94fc0818279d2f3c707ffab934db749R33

File: src/modules/message-parsers.js [hunk 2]
```diff
@@ -25,12 +31,15 @@ module.exports.in = (msg) => {
  /* eslint no-param-reassign: "off" */
 module.exports.out = (content, options) => {
   if (content !== undefined && typeof content !== 'string') {
+    if (content.error && content.error instanceof Error) {
```
Comment (mrister) [inline src/modules/message-parsers.js]: just  if (content.error instanceof Error) should be enough

**Review by jkernech — APPROVED** [review]

**Review by ykhatal — APPROVED** [review]
@Spring3 this is a nice feature :100:

**Review by webus — APPROVED** [review]

**Pull request merged**